pagerduty/resource_pagerduty_schedule.go

```diff
@@ -288,10 +329,24 @@ func expandScheduleLayers(v interface{}) ([]*pagerduty.ScheduleLayer, error) {
 	return scheduleLayers, nil
 }
 
-func flattenScheduleLayers(v []*pagerduty.ScheduleLayer) []map[string]interface{} {
+func flattenScheduleLayers(v []*pagerduty.ScheduleLayer) ([]map[string]interface{}, error) {
 	var scheduleLayers []map[string]interface{}
 
 	for _, sl := range v {
+		// A schedule layer can never be removed but it can be ended.
+		// Here we check each layer and if it has been ended we don't read it back
+		// because it's not relevant anymore.
+		if sl.End != "" {
+			end, err := timeToUTC(sl.End)
+			if err != nil {
+				return nil, err
+			}
+
+			if time.Now().UTC().After(end) {
+				continue
+			}
+		}
```

Comment:
The schema exposes an `end` attribute that users can set. This logic seems to exclude any items in a users config if that `end` time has passed, which would result in a diff as the layer is found in the config but not saved to state from the API.